src/gluonts/mx/trainer/callback.py

```diff
@@ -0,0 +1,627 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# Standard library imports
+from typing import List, Optional, Union, Dict, Any
+import logging
+
+# Third-party imports
+import numpy as np
+import mxnet.gluon.nn as nn
+import mxnet as mx
+from gluonts.core.exception import GluonTSUserError
+from gluonts.mx.trainer.model_averaging import AveragingStrategy
+from gluonts.mx.trainer.model_iteration_averaging import (
+    IterationAveragingStrategy,
+)
+from mxnet import gluon
+
+# First-party imports
+from gluonts.core.component import validated, logger
+from gluonts.mx.trainer.learning_rate_scheduler import MetricAttentiveScheduler
+from gluonts.mx.util import copy_parameters
+
+
+class Callback:
+    """
+    Abstract Callback base class.
+    Callbacks control the training of the GluonTS trainer.
+    To write a custom Callback, you can subclass Callback and overwrite one or more of the hook
+    methods. Hook methods with boolean return value stop the training if False is returned.
+    """
+
+    @validated()
+    def __init__(self, **kwargs):
+        pass
+
+    def on_train_start(self, max_epochs: int) -> None:
+        """
+        Hook that is called prior to training. This is the very first hook to be called.
+
+        Parameters
+        ----------
+        max_epochs
+            The maximum number of epochs that training is running. The actual number of epochs may
+            be fewer if another callback hook stops training early.
+        """
```

Comment:
Can we add an explicit `pass` to indicate that this method does nothing by default?

Reply:
We could, pylint complains that it is unnecessary though

Reply:
Interesting. Maybe we do `return None` then?

Reply:
That, on the other hand, lets pylint complain about unused parameters :D but I can add either, `pass` or `return None`, if you think it's better than nothing.